pixeltable/exec.py

```diff
@@ -725,10 +725,26 @@ def __init__(
 
     def _open(self) -> None:
         """Create row batch and populate with self.data"""
+
+        for info in self.input_cols:
+            assert info.col.name in self.row_column_pos
+        # before anything, convert any literal images within the input rows into references
+        _input_rows = [] # do not mutate self.input_rows, this may modify the argument also
+        for input_row in self.input_rows:
```

Comment:
... and then loop over input_cols first (like in line 747), so you only need to do the row_column_pos lookup once per column

Reply:
ok, changed.